### PR TITLE
DDBTEAM-587: Added extra no-hit matching in source tabel

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: src/Queue/SearchNoHitsProcessor.php [hunk 1]
```diff
@@ -2,18 +2,23 @@
 
 /**
  * @file
+ * Handle not-hits queue processing.
```
Comment (tuj) [inline src/Queue/SearchNoHitsProcessor.php:5]: not-hits => no-hits

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/Queue/SearchNoHitsProcessor.php (hunk 1 above)
Comment (cableman) [reply src/Queue/SearchNoHitsProcessor.php]: Fixed

**Review by tuj — APPROVED** [review]

File: src/Queue/SearchNoHitsProcessor.php [hunk 2]
```diff
@@ -119,6 +128,37 @@ public function process(Message $message, Context $session)
...
                 ]);
             }
+        } else {
+            // Try to search the data well and match source entity. This might work as there is an race between when
+            // vendors have a given cover and when the material is indexed into the data-well.
+            $type = $processMessage->getIdentifierType();
+            $material = $this->searchService->search($identifier, $type);
+
+            foreach ($material->getIdentifiers() as $is) {
+                $sourceRepos = $this->em->getRepository(Source::class);
```
Comment (turegjorup) [inline src/Queue/SearchNoHitsProcessor.php:138]: `$sourceRepos`should be called before the loop, not inside it

File: src/Queue/SearchNoHitsProcessor.php [hunk 3]
```diff
@@ -119,6 +128,37 @@ public function process(Message $message, Context $session)
...
+                $source = $sourceRepos->findOneBy([
+                    'matchId' => $is->getId(),
+                    'matchType' => $is->getType(),
+                ]);
```
Comment (turegjorup) [inline src/Queue/SearchNoHitsProcessor.php:142]: Theoretically we can have multiple sources for any `id`/`type` pair. We should load them all and use the one from the best ranked vendor

**Review by turegjorup — CHANGES_REQUESTED** [review]

File: src/Queue/SearchNoHitsProcessor.php (hunk 2 above)
Comment (cableman) [reply src/Queue/SearchNoHitsProcessor.php:138]: Fixed

File: src/Queue/SearchNoHitsProcessor.php (hunk 3 above)
Comment (cableman) [reply src/Queue/SearchNoHitsProcessor.php:142]: Fixed

**Review by turegjorup — APPROVED** [review]

**Pull request merged**